augur/validate.py

```diff
@@ -163,7 +169,21 @@ def auspice_config_v2(config_json, **kwargs):
     validate(config, schema, config_json)
 
 def export_v2(main_json, **kwargs):
-    main_schema = load_json_schema("schema-export-v2.json")
+    # The main_schema uses references to other (local) JSONs so we need to
+    # define the mapping to use here. Because the main_schema uses the $id
+    # "https://nextstrain.org/schemas/dataset/v2" this is combined with the
+    # value we use for $ref to create a URI, and it is this URI we need to
+    # specify here so that the local file is used. If we don't provide this
+    # mapping then we'll attempt to fetch this URI, and nextstrain.org will
+    # return 404 with a JSON response "{'error': 'Not Found'}"... and jsonschema
+    # will treat that as a valid schema and proceed. For development purposes I
+    # suggest you remove the $id from the main schema to avoid remote schema
+    # requests. Perhaps one day we'll parse this straight from the schema
+    # itself.                                               james August 2023
+    refs = {
+        'https://nextstrain.org/schemas/dataset/genome_annotations': "schema-annotations.json"
+    }
+    main_schema = load_json_schema("schema-export-v2.json", refs)
```

Comment:
Per my other comment, we'd still provide the validator here with a local provisioning of the schemas by their id URLs to avoid the fetch and use the local (possibly modified) copies.

```python
    refs = {
        'https://nextstrain.org/schemas/augur/annotations': "schema-annotations.json"
    }
```

The upshot again is that the references would still be valid when processed outside of Augur.

Reply:
Done